cmd/podman/run.go

```diff
@@ -61,6 +63,19 @@ func runCmd(c *cliconfig.RunValues) error {
 		return errors.Wrapf(err, "error creating libpod runtime")
 	}
 	defer runtime.DeferredShutdown(false)
+	defer func() {
+		if c.Bool("rmi") {
```

Comment:
I don't think this is the full solution - it won't work if the user detaches from the container with the detach keys while it's still running. We'll need changes to `podman container cleanup` to add the `rmi` functionality there, and then we can pass `--rmi` in by changing the command we use to invoke `podman container cleanup`  via `createExitCommand()` in `pkg/spec/createconfig.go`

Reply:
Done. I hope I didn't miss anything and I understood you correctly.